File: internal/content/testdata/bench/csharp/nc-jump-game-ii.cs
public class Solution {
    public long jump(long[] nums) {
        long jumps = 0;
        long curEnd = 0;
        long farthest = 0;
        for (int i = 0; i < nums.Length - 1; i++) {
            farthest = Math.Max(farthest, i + nums[i]);
            if (i == curEnd) {
                jumps++;
                curEnd = farthest;
            }
        }
        return jumps;
    }
}

File: internal/content/testdata/bench/rust/nc-jump-game-ii.rs
fn jump(nums: Vec<i64>) -> i64 {
    let mut jumps = 0i64;
    let mut cur_end = 0i64;
    let mut farthest = 0i64;
    let len = nums.len();
    if len == 0 {
        return 0;
    }
    for i in 0..(len - 1) {
        let reach = i as i64 + nums[i];
        if reach > farthest {
            farthest = reach;
        }
        if i as i64 == cur_end {
            jumps += 1;
            cur_end = farthest;
        }
    }
    jumps
}
